dev/BackgroundTask/BackgroundTaskBuilder/BackgroundTaskBuilder.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "BackgroundTaskBuilder.h"
+#include "Microsoft.Windows.ApplicationModel.Background.BackgroundTaskBuilder.g.cpp"
+#include <winrt/Windows.ApplicationModel.Activation.h>
+
+namespace winrt
+{
+    using namespace winrt::Windows::ApplicationModel::Background;
+    using namespace Windows::Storage;
+}
+
+namespace winrt::Microsoft::Windows::ApplicationModel::Background::implementation
+{
+    void BackgroundTaskBuilder::SetTaskEntryPointClsId(winrt::guid clsId)
+    {
+        m_taskEntryPointClsId = clsId;
+    }
+
+    void BackgroundTaskBuilder::Name(winrt::hstring name)
+    {
+        m_name = name;
+    }
+
+    void BackgroundTaskBuilder::SetTrigger(winrt::IBackgroundTrigger trigger)
+    {
+        m_trigger = trigger;
+    }
+
+    winrt::BackgroundTaskRegistration BackgroundTaskBuilder::Register()
+    {
+        winrt::BackgroundTaskBuilder builder;
+        builder.Name(m_name);
+        builder.SetTrigger(m_trigger);
+        builder.TaskEntryPoint(L"Microsoft.Windows.ApplicationModel.Background.UniversalBGTask.Task");
+        winrt::BackgroundTaskRegistration taskRegistration = builder.Register();
+
+        winrt::ApplicationDataContainer localSettings = winrt::ApplicationData::Current().LocalSettings();
```

Comment:
The background task will work for both packaged and unpackaged apps? Please check ApplicationData::Current() works for un-packaged app.

Reply:
As of now it will support only packaged apps, If we need to support unpackaged apps we can support them in later iterations

Reply:
@godlytalias there is a merged pr for ApplicationData (GetForUnpackaged), however it seems that it marked as contract v2, so mybe you can consider it.
https://github.com/microsoft/WindowsAppSDK/pull/4057

Reply:
Thanks @ghost1372 for pointing out that API. I will need to see the working of background task feature with unpackaged apps and then add the support in an upcoming PR